pkg/operators/process/helpers/processhelpers.go

```diff
@@ -0,0 +1,387 @@
+			bufPool.Put(xbuf)
+			return nil, 0, fmt.Errorf("reading file: %w", e)
+		}
+	}
+	return xbuf, n, nil
+}
+
+func GetProcessInfo(pid int, timeDelta float64, options Options) (ProcessInfo, error) {
+	pi := ProcessInfo{PID: pid}
+
+	// read status file to extract requested data
+	if needStatus(options) {
+		path := filepath.Join(host.HostProcFs, strconv.Itoa(pid), "status")
+		s, err := os.Open(path)
+		if err != nil {
+			return pi, fmt.Errorf("opening status file for pid %d: %w", pid, err)
+		}
+		xbuf, n, err := readBytes(s)
+		if err != nil {
+			return pi, fmt.Errorf("reading status file for pid %d: %w", pid, err)
+		}
+		s.Close()
+		data := (*xbuf)[:n]
+
+		bComm := options.WithComm()
+		bPPID := options.WithPPID()
+		bState := options.WithState()
+		bUid := options.WithUID()
+		bVmSize := options.WithVmSize()
+		bVmRSS := options.WithVmRSS()
+		bThreads := options.WithThreadCount()
+
+		if options.WithMemoryRelative() {
+			bVmRSS = true
+		}
+
+		for i := 0; i < len(data); {
+			j := bytes.IndexByte(data[i:], '\n')
+			if j < 0 {
+				j = len(data) - i
+			}
+			line := data[i : i+j]
+
+			switch {
+			case bComm && bytes.HasPrefix(line, prefName):
+				bComm = false
+				pi.Command = unescapeCommandBytes(line[len(prefName):])
+			case bPPID && bytes.HasPrefix(line, prefPPid):
+				bPPID = false
+				pi.PPID = int(parseTrimDecimal(line[len(prefPPid):]))
+			case bState && bytes.HasPrefix(line, prefState):
+				bState = false
+				if len(line) > len(prefState) {
+					pi.State = string(line[len(prefState)])
+				}
+			case bUid && bytes.HasPrefix(line, prefUid):
+				bUid = false
+				pi.Uid = uint32(parseTrimDecimal(line[len(prefUid):]))
+			case bVmSize && bytes.HasPrefix(line, prefVmSize):
+				bVmSize = false
+				pi.MemoryVirtual = parseTrimDecimal(line[len(prefVmSize):]) * 1024
+			case bVmRSS && bytes.HasPrefix(line, prefVmRSS):
+				bVmRSS = false
+				pi.MemoryRSS = parseTrimDecimal(line[len(prefVmRSS):]) * 1024
+			case bThreads && bytes.HasPrefix(line, prefThreads):
+				bThreads = false
+				pi.ThreadCount = int(parseTrimDecimal(line[len(prefThreads):]))
+			}
+
+			if !bComm && !bPPID && !bState && !bUid && !bVmSize && !bVmRSS && !bThreads {
+				// exit early if we got everything we need
+				break
+			}
+
+			i += j + 1
+		}
+		bufPool.Put(xbuf)
+	}
+
+	if options.WithMemoryRelative() {
+		totalMem := options.TotalMemory()
+		if totalMem > 0 {
+			pi.MemoryRelative = 100 * float64(pi.MemoryRSS) / float64(totalMem)
+		}
+	}
+
+	// read stat file to extract CPU Usage & start time
+	if needStat(options) {
+		path := filepath.Join(host.HostProcFs, strconv.Itoa(pid), "stat")
+		s, err := os.Open(path)
+		if err != nil {
+			return pi, fmt.Errorf("open stat %d: %w", pid, err)
+		}
+		xbuf, n, err := readBytes(s)
+		if err != nil {
+			return pi, fmt.Errorf("reading stat file for pid %d: %w", pid, err)
+		}
+		s.Close()
+		data := (*xbuf)[:n]
+
+		o := bytes.IndexByte(data, '(')
+		c := bytes.LastIndexByte(data, ')')
```

Comment:
JFYI, I referenced this code in:
https://github.com/inspektor-gadget/inspektor-gadget/pull/4520#discussion_r2111569105

Reply:
That was faster than expected ;) Moved the helpers out of the operators dir, directly into /pkg.

/cc @mqasimsarfraz